crypto/ctype.c

```diff
@@ -278,3 +290,69 @@ int ossl_ascii_isdigit(const char inchar) {
         return 1;
     return 0;
 }
+
+/* str[n]casecmp_l is defined in POSIX 2008-01. Value is taken accordingly
+ * https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html */
+
+#if (defined OPENSSL_SYS_WINDOWS) || (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200809L)
+
+# if defined OPENSSL_SYS_WINDOWS
+# define locale_t _locale_t
+# define freelocale _free_locale
+# define strcasecmp_l _stricmp_l
+# define strncasecmp_l _strnicmp_l
+# endif
+static locale_t loc;
+
+# ifndef FIPS_MODULE
+void *OPENSSL_C_locale() {
```

Comment:
This function does not have to be and isn't exported. Please rename it to `ossl_c_locale`.

Reply:
It isn't exported, though it may make sense to have it exported - if it returns NULL, we don't have a locale-agnostic comparison... 

Will change, thanks!

Reply:
Done